services/worker/pyproject.toml

```diff
@@ -9,12 +9,12 @@ license = "Apache-2.0"
 Pillow = "^9.4.0"
 PyICU = "^2.10.2"
 aiohttp = "^3.8.4"
-apache-beam = "2.41.0" # ^2 gives a InvalidWheelName error because it tries to install 2.42 that has not been released...
```

Comment:
We don't need apache-beam for the datasets-server, but some dataset scripts require it. It would be a regression for these... I understand that we remove it here to allow upgrading pyarrow to a recent version.

It's hard to have an idea of how many datasets require apache-beam. Do we have some script that analyzes the content of all the dataset scripts on the Hub @huggingface/datasets?

Another solution would be to have apache-beam work with a recent version of pyarrow. But, as https://github.com/huggingface/datasets/issues/5430 is not closed yet, I imagine that it's not currently possible.

Reply:
You can ignore apache beam, only wikipedia and natural_questions need it. It would be easier to switch them to non-beam datasets that supporting beam datasets x)

Reply:
OK. So: 
- https://huggingface.co/datasets/wikipedia
- https://huggingface.co/datasets/natural_questions

Anyway: none of them is currently working. I will open a discussion on them to remove dependency to apache_beam.

Reply:
- https://huggingface.co/datasets/wikipedia/discussions/9
- https://huggingface.co/datasets/natural_questions/discussions/4

Reply:
I have created a discussion in "wiki40b" as well:
- https://huggingface.co/datasets/wiki40b/discussions/2